src/Sylius/Bundle/CoreBundle/Form/Type/Checkout/ShipmentType.php

```diff
@@ -50,25 +50,15 @@ public function __construct($dataClass, TranslatorInterface $translator)
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
-        $criteria = $options['criteria'];
-        $channel = $options['channel'];
-
-        $notBlank = new NotBlank(['groups' => ['sylius']]);
-        $notBlank->message = $this->translator->trans('sylius.checkout.shipping_method.not_blank', [], 'validators');
-
         $builder
-            ->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) use ($notBlank, $criteria, $channel) {
+            ->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) {
```

Comment:
Unfortunately, ``addEventSubscriber`` method requires ``EventSubscriber`` object as an argument. As it's legacy shipping step type and will be probably removed in the nearest future, I would leave it as it is.